Entity/Type/SafeObjectType.php

```diff
@@ -2,17 +2,17 @@
 
 namespace JMS\JobQueueBundle\Entity\Type;
 
-use Doctrine\DBAL\Types\ObjectType;
+use Doctrine\DBAL\Types\JsonType;
 
-class SafeObjectType extends ObjectType
+class SafeObjectType extends JsonType
```

Comment:
This causes some trouble. We currently have data in the database and that is stored (in Application:87) by serializing a `FlattenException`. That means that jobs without an error always have `N;` as the value, and jobs with an error a mega long hex value that looks something like `0x4F3A35373A2253796D666F6E795C436F6D706F6....`
This of course can't be handled by the JsonType, so the worker do not start if there are existing jobs.

So, we are dealing with two problems here:
1. the value in the database has to become a JSON for existing jobs
2. the value in the database has to become a JSON for new jobs

And a third one:
3. This has to be done ideally in a backwards compatible way so that we can migrate back if we need to and ideally have the old and the new workers running at the same time (probably not possible)

The first problem can probably be solved by a migration. But that is not backards-compatible then. So, maybe we have to consider to add a new column.

I tried json_encode for the `FlattenException`, but unfortunately that just returns an empty JSON object. I was not able yet to find another way to get a proper JSON from nested exceptions. I think that we maybe have to consider just saving the stacktrace as string, but then we lose some valuable information, like the exception message.

I'll try to create a JsonSerializable FlattenException; that should solve that problem. And I am thinking of either rewriting the JobSafeObject to handle both - the old and the new - style in the database. But then we still have to perform a double deployment. I think it might be easier to write a migration that rewrites the stuff in the database, but I fear that this is can't be reversed. So the better solution is probably to introduce a new column and just set the existing one always to `N;` until we can finally remove / rename it.

Reply:
As far as I remember, that's a change that I had to make. I didn't stumble across those implications during my testing. So my current status is that you don't need help with this (anymore)?

Reply:
yes exactly; I have pushed the updated code with all my fixes that were required (see also my other comments)

Reply:
> I didn't stumble across those implications during my testing.

Probably because you didn't have old jobs in your database that were already executed. The stacktrace is only written when a job has been started and not when the job is still pending. So, this is an edge case that our tests can not really cover and I also only discovered this when I tried to activate the workers on Staging